genic/main.c

```diff
@@ -43,7 +43,7 @@ int main(int argc, char **argv)
 
   petapm_init(All.BoxSize, All.Nmesh, All.NumThreads);
   /*Initialise particle spacings*/
-  const double meanspacing = All.BoxSize / All2.Ngrid;
+  const double meanspacing = All.BoxSize / DMAX(All2.Ngrid, All2.NgridGas);
```

Comment:
This looks strange. perhaps need meanspacing for each species instead.

Reply:
It is only used for computing the shifts - the important thing is actually that the shift is less than one grid cell for all particles, so this is the right thing to do.